src/devices/src/virtio/vsock/device.rs

```diff
@@ -148,10 +148,19 @@ where
                 Ok(mut pkt) => {
                     if self.backend.recv_pkt(&mut pkt, mem).is_ok() {
                         match pkt.commit_hdr(mem) {
-                            Ok(()) => VSOCK_PKT_HDR_SIZE as u32 + pkt.len(),
+                            Ok(()) => pkt.len().checked_add(VSOCK_PKT_HDR_SIZE as u32).unwrap_or_else(|| {
+                                warn!(
+                                    "vsock: Packet length greater than maximum expected size: {} > {}.\
+                                     Discarding the package.",
+                                    pkt.len(),
+                                    u32::MAX - VSOCK_PKT_HDR_SIZE as u32
+                                );
+                                0
```

Comment:
I have some concerns here related to value we are returning (i.e 0) given that we write the header to memory (commit_hdr) even though the packet length + header size exceeds u32::MAX. Let me wrap my head around it and come back with a conclusion.

Reply:
How about we check that addition would not overflow (i.e. `pkt.len() < u32::MAX - VSOCK_PKT_HDR_SIZE`) before we commit the header to memory. If true, then we can write the packet to guest. Otherwise, we return 0 and drop the package. WDYT?

Reply:
Yes, so it is actually correct to return 0 but there is still a problem with the code. We need to do the overflow check before committing the header (i.e writing the header to memory) since if we are in the scenario where we have an overflow this might mean we do not have enough space to write the header. 
So, my suggestion is to move the check in `commit_hdr` function and return an error if there is overflow.

Reply:
> How about we check that addition would not overflow (i.e. `pkt.len() < u32::MAX - VSOCK_PKT_HDR_SIZE`) before we commit the header to memory. If true, then we can write the packet to guest. Otherwise, we return 0 and drop the package. WDYT?

Indeed, this was my suggestion also, saw your comment after replying.